### PR TITLE
New Package Chooser Dialog

### PR DESCRIPTION
**Pull request by onovotny**
PR to review/merge in the new package chooser dialog that @khaister contributed.

Would appreciate a review from @campersau as well.

### CODE REVIEW

**Comment by onovotny** [comment]
Observations: one thing this needs is the reserved package prefix checkbox.

**Comment by onovotny** [comment]
Seems like https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/blob/f87c7faa7d6f20c8eb4127854d50964f5b74aeb9/PackageViewModel/PackageChooser/PackageInfoViewModel.cs#L201 is returning `null` for published. Will investigate separately.

File: PackageExplorer/app.config [hunk 1]
```diff
@@ -2,10 +2,70 @@
 <configuration>
     <configSections>
         <sectionGroup name="userSettings" type="System.Configuration.UserSettingsGroup, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089">
+            <section name="PackageExplorer.Properties.Settings" type="System.Configuration.ClientSettingsSection, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" allowExeDefinition="MachineToLocalUser" requirePermission="false" />
             <section name="NuGetPackageExplorer.Properties.Settings" type="System.Configuration.ClientSettingsSection, System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" allowExeDefinition="MachineToLocalUser" requirePermission="false" />
```
Comment (campersau) [inline PackageExplorer/app.config]: I think the `NuGetPackageExplorer` settings should be removed. See https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/pull/404#discussion_r183957221

---

The settings were originally in the `PackageExplorer` namespace (https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/commit/b698085a15608b978c6b63db4fbf55c68fba187d#diff-0af6157b86a5943f59ec5be95d647bb7) but this got changed in https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/commit/b1729afb016588acf6b6f75959f6b65bcd60b519#diff-0af6157b86a5943f59ec5be95d647bb7 and with https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/commit/d9d37f8ff097299b11c0a2e0ffb966d7d14d8d72#diff-0af6157b86a5943f59ec5be95d647bb7 it got changed back to `PackageExplorer` but without updating the `app.config` file.

**Comment by campersau** [comment]
@khaister Ups, I just did these. See #409 and #411

**Comment by onovotny** [comment]
sorry, did a force push on this branch and master to clean up my mess :/

**Pull request merged**